packages/stream-metadata/src/routes/spaceMetadata.ts

```diff
@@ -1,36 +1,138 @@
+import { ChunkedMedia } from '@river-build/proto'
 
 import { isValidEthereumAddress } from '../validators'
+import { getSpaceDapp } from '../contract-utils'
+import { config } from '../environment'
+import { getStream } from '../riverStreamRpcClient'
+import { getSpaceImage } from './spaceImage'
 
-export function fetchSpaceMetadata(
+export interface SpaceMetadataResponse {
+	name: string
+	description: string
+	image: string | undefined
+}
+
+export async function fetchSpaceMetadata(
 	request: FastifyRequest,
 	reply: FastifyReply,
 	serverUrl: string,
 ) {
 	const logger = request.log.child({ name: fetchSpaceMetadata.name })
 	const { spaceAddress } = request.params as { spaceAddress?: string }
-	logger.info({ spaceAddress }, 'GET /space/../metadata')
 
 	if (!spaceAddress) {
+		logger.error('spaceAddress parameter is required')
 		return reply
 			.code(400)
 			.send({ error: 'Bad Request', message: 'spaceAddress parameter is required' })
 	}
 
 	// Validate spaceAddress format using the helper function
 	if (!isValidEthereumAddress(spaceAddress)) {
+		logger.error({ spaceAddress }, 'Invalid spaceAddress format')
 		return reply
 			.code(400)
 			.send({ error: 'Bad Request', message: 'Invalid spaceAddress format' })
 	}
 
-	const dummyJson = {
-		name: '....',
-		description: '....',
-		members: 99999,
-		fees: '0.001 eth',
-		image: `${serverUrl}/space/${spaceAddress}/image`,
+	const spaceDapp = getSpaceDapp()
+	let spaceInfo: SpaceInfo | undefined
+	try {
+		spaceInfo = await spaceDapp.getSpaceInfo(spaceAddress)
+	} catch (error) {
+		logger.error({ spaceAddress, error }, 'Failed to fetch space contract info')
+		return reply
+			.code(404)
+			.send({ error: 'Not Found', message: 'Failed to fetch space contract info' })
+	}
+
+	if (!spaceInfo) {
+		logger.error({ spaceAddress }, 'Space contract not found')
+		return reply.code(404).send({ error: 'Not Found', message: 'Space contract not found' })
+	}
+
+	const spaceMetadata: SpaceMetadataResponse = {
+		name: spaceInfo.name,
+		description: getSpaceDecription(spaceInfo),
+		image: await getImageUrl(logger, spaceInfo.uri, spaceAddress),
 	}
 
-	return reply.header('Content-Type', 'application/json').send(dummyJson)
+	return reply.header('Content-Type', 'application/json').send(spaceMetadata)
+}
+
+function getSpaceDecription({ shortDescription, longDescription }: SpaceInfo): string {
+	if (shortDescription && longDescription) {
+		return `${shortDescription}\n\n${longDescription}`
+	}
+
+	if (shortDescription) {
+		return shortDescription
+	}
+
+	return longDescription || ''
+}
+
+async function getImageUrl(logger: FastifyBaseLogger, contractUri: string, spaceAddress: string) {
+	const hasSpaceImageExist = await hasSpaceImage(logger, spaceAddress)
+	if (!hasSpaceImageExist) {
+		return undefined
+	}
+
+	const isDefaultPort =
+		config.riverStreamMetadataBaseUrl.port === '' ||
+		config.riverStreamMetadataBaseUrl.port === '80' ||
+		config.riverStreamMetadataBaseUrl.port === '443'
```

Comment:
Why are you checking ports here? Why are building an image url here at all?

Reply:
Is this implementing an opensea spec? if so can you link it in the pr description?

Reply:
no this is for localhost dev only. I'll simplify it by checking config.riverEnv instead.